server/genserver.go

```diff
@@ -317,9 +319,24 @@ type handlerOp struct {
 	SingleStringPathParameter        bool
 	SingleStringPathParameterVarName string
 	StatusCodeToType                 map[int]string
+	HasUserContext                   bool
 }
 
 var handlerTemplate = `
+{{define "userContextExtraction"}}
+		// Extract UserContext as a separate parameter
+		var xUserContext *models.UserContext
+		xUserContextStr := r.Header.Get("X-User-Context")
+		if len(xUserContextStr) > 0 {
+			var uc models.UserContext
+			if err := json.Unmarshal([]byte(xUserContextStr), &uc); err != nil {
+				logger.FromContext(ctx).AddContext("error", err.Error())
+				http.Error(w, jsonMarshalNoError({{index .StatusCodeToType 400}}{Message: fmt.Sprintf("failed to unmarshal X-User-Context header: %v", err)}), http.StatusBadRequest)
+				return
+			}
+			xUserContext = &uc
+		}
+{{end}}
```

Comment:
this code template is reused later in the file whenever an endpoint's server code needs to extract UserContext from a header and convert it from a JSON string to a UserContext struct. This is potentially an area where we'd want to use the rbac-scoping-utils helper function that checks if the value is an empty string, but I'm not sure how tricky that would be to include in the autogenerated code.

A potential alternate solution is to just use this code (or something similar) to JSON unmarshal the header, but in the controller code, if the UserContext arg is nil, then we can assume the header wasn't passed in.